File: frontend/src/types/guardrail.ts
export type GuardrailCategory =
  | "Prompt Validation"
  | "PII Detection"
  | "Compliance"
  | "Risk Rules"
  | "Fraud Rules"
  | "Product Eligibility"
  | "Investment Restrictions"
  | "Human Approval"

export interface Guardrail {
  id: string
  category: GuardrailCategory
  name: string
  description: string
  status: "Enabled" | "Disabled"
  severity: "Low" | "Medium" | "High" | "Critical"
  violationsCount: number
  countries: string[]
}

File: backend/app/schemas/guardrail.py
from typing import Literal
from pydantic import BaseModel

GuardrailCategory = Literal[
    "Prompt Validation",
    "PII Detection",
    "Compliance",
    "Risk Rules",
    "Fraud Rules",
    "Product Eligibility",
    "Investment Restrictions",
    "Human Approval",
]


class Guardrail(BaseModel):
    id: str
    category: GuardrailCategory
    name: str
    description: str
    status: Literal["Enabled", "Disabled"]
    severity: Literal["Low", "Medium", "High", "Critical"]
    violationsCount: int
    countries: list[str]
